packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_type3_staticprofile.cpp

```diff
@@ -0,0 +1,322 @@
+  //
+  RCP<TimeMonitor> timerElementLoopMatrix = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("3) ElementLoop  (Matrix)")));
+
+  RCP<MatrixType> crs_matrix = rcp(new MatrixType(crs_graph));
+
+  scalar_2d_array_type element_matrix;
+  Kokkos::resize(element_matrix, 4, 4);
+
+  Teuchos::Array<GlobalOrdinal> column_global_ids(4);     // global column ids list
+  Teuchos::Array<Scalar> column_scalar_values(4);         // scalar values for each column
+
+  // Loop over owned elements:
+  for(size_t element_gidx=0; element_gidx<mesh.getNumOwnedElements(); element_gidx++)
+  {
+    // Get the stiffness matrix for this element
+    ReferenceQuad4(element_matrix);
+
+    // Fill the global column ids array for this element
+    for(size_t element_node_idx=0; element_node_idx<owned_element_to_node_ids.extent(1); element_node_idx++)
+    {
+      column_global_ids[element_node_idx] = owned_element_to_node_ids(element_gidx, element_node_idx);
+    }
+    
+    // For each node (row) on the current element:
+    // - populate the values array
+    // - add values to crs_matrix if the row is owned.
+    //   Note: hardcoded 4 here because we're using quads.
+    for(size_t element_node_idx=0; element_node_idx<4; element_node_idx++)
+    { 
+      GlobalOrdinal global_row_id = owned_element_to_node_ids(element_gidx, element_node_idx);
+      if(mesh.nodeIsOwned(global_row_id)) 
+      {
+        for(size_t col_idx=0; col_idx<4; col_idx++)
+        {
+          column_scalar_values[col_idx] = element_matrix(element_node_idx, col_idx);
+        }
+        crs_matrix->sumIntoGlobalValues(global_row_id, column_global_ids, column_scalar_values);
+      }
+    }
+  }
+
+  // Loop over ghost elements:
+  // - This loop is the same as the element loop for owned elements, but this one
+  //   is for ghost elements.
+  for(size_t element_gidx=0; element_gidx<mesh.getNumGhostElements(); element_gidx++)
+  {
+    ReferenceQuad4(element_matrix);
+
+    for(size_t element_node_idx=0; element_node_idx<ghost_element_to_node_ids.extent(1); element_node_idx++)
+    {
+      column_global_ids[element_node_idx] = ghost_element_to_node_ids(element_gidx, element_node_idx);
+    }
+    
+    for(size_t element_node_idx=0; element_node_idx<4; element_node_idx++)
+    { 
+      GlobalOrdinal global_row_id = ghost_element_to_node_ids(element_gidx, element_node_idx);
+      if(mesh.nodeIsOwned(global_row_id)) 
+      {
+        for(size_t col_idx=0; col_idx<4; col_idx++)
+        {
+          column_scalar_values[col_idx] = element_matrix(element_node_idx, col_idx);
+        }
+        crs_matrix->sumIntoGlobalValues(global_row_id, column_global_ids, column_scalar_values);
+      }
+    }
+  }
+  timerElementLoopMatrix = Teuchos::null;
+
+  // After the contributions are added, 'finalize' the matrix using fillComplete()
+  {
+    RCP<TimeMonitor> timerFillCompleteMatrix = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("4) FillComplete (Matrix)")));
```

Comment:
MueLu has a few places that maybe perhaps could possibly need `RCP<TimeMonitor>`, but this place pretty egregiously does not ;-) .

Reply:
I would say, keep the timings, but don't do `RCP<TimeMonitor>`.  You already have a scope; let the scope guard be a scope guard ;-) .  Do it like this:
```
{
  TimeMonitor time_f (*TimeMonitor::getNewTimer ("f"));
  f();
}
{
  TimeMonitor time_g (*TimeMonitor::getNewTimer ("g"));
  g();
}
```

Reply:
Ahhh... ok.  I misunderstood your comment before.  Thought you wanted me to clear out the timings.

Yeah, using both scope and smart-pointer isn't necessary.  
<-------- n00b!

Reply:
ah, ok.  That makes sense then why I'd be using that pattern since the framework for these was put together for me by Chris S.   There are probably a lot of MueLu-isms in it that I won't know if they're normative or not. ;)